### DISCUSSION TITLE
How to silence Bash Operator and optimize memory usage?

### DISCUSSION THREAD

**Discussion by kiranzo** [apache/airflow#34524] [Q&A] [answered]
So, I'm calling a Java Jar tool in my data processing pipeline, and it spams to the standard output A LOT, all of that ends up with INFO status in airflow console. 
The test set I have now creates 184 bash operator instances, and there will be more when the pipeline goes live. Airflow periodically sends SIGTERM to some of these instances due to lack of memory, so I set `max_active_tasks_per_dag` in the airflow.cfg to 4, but I still keep getting it.
I suspect it might be a little bit better if I manage to stop the Jar from spamming into Airflow console and just make it dump everything into a log file.
I experimentally tried to change my code for BashOperator like this:

`java -jar /path/to/tool.jar ${INPUT_PATH} ${OUTPUT_PATH} > ${INPUT_PATH}.log`

but it just created empty ${INPUT_PATH}.log file and kept spamming into airflow task instance output with INFO status.

Is there a way to redirect output directly to file? I suspect that spam eats a lot of memory, and there is no way to turn it off in Jar options.

**Comment by Taragolis** [comment, marked as answer]
I guess the problem that logger write output into the `stderr` instead of `stdout` which are common practice.
So you could try to redirect both stdout/stderr by use `&>` instead of `>`, or you could redirect only stderr in file by use `2>` instead of `>`

**Reply by kiranzo** [reply]
Thanks! I changed the command for bash operator like this

`java -jar /path/to/tool.jar ${INPUT_PATH} ${OUTPUT_PATH} > ${INPUT_PATH}.log  2>&1`

and it redirects everything to file now. All that spam was indeed coming from the stderr output.
